astropy/cosmology/parameter/tests/test_descriptors.py

```diff
@@ -3,15 +3,70 @@
 """Testing :mod:`astropy.cosmology.parameter._descriptor`."""
 
 from types import MappingProxyType
+from typing import ClassVar
 
 import pytest
 
 from astropy.cosmology._utils import all_cls_vars
 from astropy.cosmology.parameter import Parameter
+from astropy.cosmology.parameter._descriptors import ParametersAttribute
+
+
+class Obj:
+    a: ClassVar = 1
+    b: ClassVar = 2
+    c: ClassVar = 3
+    _attr_map: ClassVar = ("a", "b", "c")
+
+    attr = ParametersAttribute(attr_name="_attr_map")
```

Comment:
Might it be better without the indirection? Maybe it is clearer to just reinitialize `ParametersAttribute` in sublasses instead of adding to `_attr_map`. One could even define `|` or `+` to allow adding to what was already there in a superclass (I do this in my baseband package, https://github.com/mhvk/baseband/blob/90a033157cabe30f14ef1a1230c19e00689f736b/baseband/vdif/header.py#L595-L598)

Reply:
> Maybe it is clearer to just reinitialize ParametersAttribute in sublasses instead of adding to _attr_map?

I like this idea. I think it's worth exploring in a followup PR! I want to keep this PR about fixing the bug and adding tests to ensure it's ilk don't reappear.

Reply:
Yes, I realize I got rather off track in my review!